### PR TITLE
Densityplot: add support for discrete variables

### PR DESCRIPTION
**Pull request by aloctavodia**

### CODE REVIEW

**Comment by ColCarroll** [comment]
Can you post a screen shot/example of what this looks like?  Maybe also a test (our plotting "tests" are just making sure you can call the functions)?

**Comment by aloctavodia** [comment]
In this example sigma is continuous and gamma discrete

density_plot


 Densityplot is already included in our tests (see #2741)

**Comment by aloctavodia** [comment]
ups! I just notice a problem (with the xticks)

File: pymc3/plots/densityplot.py [hunk 1]
```diff
@@ -145,34 +148,43 @@ def _kde_helper(vec, vname, c, bw, alpha, point_estimate, hpd_markers, outline,
...
+
+        if shade:
+            ax.fill_between(x, density, where=cut, color=c, alpha=shade)
+
+    else:
+        xmin, xmax = hpd(vec, alpha)
+        bins = range(xmin, xmax+1)
+        if outline:
+            ax.hist(vec, bins=bins, color=c, histtype='step')
+        ax.hist(vec, bins=bins, color=c, alpha=shade)
```
Comment (junpenglao) [inline pymc3/plots/densityplot.py:175]: Better to set `density=None` to True.

**Comment by aloctavodia** [comment]
This is how it looks like with the fix

density_plot

File: pymc3/plots/densityplot.py (hunk 1 above)
Comment (aloctavodia) [reply pymc3/plots/densityplot.py:175]: That will only change the scale of y. But that scale is not plotted, the yticks are set to an empty list.

Comment (junpenglao) [reply pymc3/plots/densityplot.py:175]: I see. Fine with me.

File: pymc3/plots/densityplot.py [hunk 2]
```diff
@@ -64,7 +65,7 @@ def densityplot(trace, models=None, varnames=None, alpha=0.05, point_estimate='m
 
     """
     if point_estimate not in ('mean', 'median', None):
-        raise ValueError("Point estimate should be 'mean' or 'median'")
+        raise ValueError("Point Estimate should be 'mean', 'median' or None")
```
Comment (twiecki) [inline pymc3/plots/densityplot.py]: why `Estimate`?

**Comment by twiecki** [comment]
This looks great!

File: pymc3/plots/densityplot.py (hunk 2 above)
Comment (aloctavodia) [reply pymc3/plots/densityplot.py]: just a typo

**Pull request merged**